### PR TITLE
gh-124487: Require at least Windows 10 SDK and for install

### PR DESCRIPTION
**Pull request by zooba**
* Issue: gh-124487

### CODE REVIEW

**Comment by zooba** [comment]
#124676 is an alternative fix that might be considered for 3.13 instead of this one.

File: PC/pyconfig.h.in [hunk 1]
```diff
@@ -170,8 +170,8 @@ WIN32 is still required for the locale module.
 
 /* set the version macros for the windows headers */
 /* Python 3.9+ requires Windows 8 or greater */
```
Comment (Wulian233) [inline PC/pyconfig.h.in]:
```suggestion
/* Python 3.13+ requires Windows 10 or greater */
```

**Review by Wulian233 — COMMENTED** [review]
Is the minimum installation requirement for PR now Win10? If yes, I think the `FailureOldOS` information needs to be changed as well

https://github.com/search?q=repo%3Apython%2Fcpython%20FailureOldOS&type=code

`At least Windows 8.1 or Windows Server 2012 are required to install`

**Comment by zooba** [comment]
Thanks for the review! Good catches

**Pull request merged**